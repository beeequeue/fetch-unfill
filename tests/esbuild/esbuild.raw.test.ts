import path from "node:path"

import esbuild from "esbuild"
import fetchUnfillAliases from "fetch-unfill/aliases"
import { describe, expect, it } from "vitest"

import { createTester } from "../utils.js"

const test = createTester(
  "esbuild",
  async (name: string, extraOptions: esbuild.BuildOptions | null = {}) => {
    const options = {
      write: false,
      nodePaths: [path.resolve(__dirname, "..", "node_modules")],

      platform: "node",
      target: "node18",
      bundle: true,
      minify: true,
    } satisfies esbuild.BuildOptions

    const cjs = await esbuild.build({
      ...options,
      ...extraOptions,

      entryPoints: [path.resolve(__dirname, "..", "fixtures", `${name}.cjs`)],
      format: "cjs",
    })

    const mjs = await esbuild.build({
      ...options,
      ...extraOptions,

      entryPoints: [path.resolve(__dirname, "..", "fixtures", `${name}.mjs`)],
      format: "esm",
    })

    expect(cjs.errors).toStrictEqual([])
    expect(cjs.outputFiles).toHaveLength(1)
    expect(mjs.errors).toStrictEqual([])
    expect(mjs.outputFiles).toHaveLength(1)

    return {
      cjs: cjs.outputFiles![0].text,
      mjs: mjs.outputFiles![0].text,
    }
  },
)

const alias = {
  alias: fetchUnfillAliases,
} satisfies esbuild.BuildOptions

describe("node-fetch", () => {
  it("bundles the original package", async () => {
    await test("node-fetch", null)
  })

  it("unfills it", async () => {
    await test("node-fetch", alias, true)
  })
})

describe("cross-fetch", () => {
  it("bundles the original package", async () => {
    await test("cross-fetch", null)
  })

  it("unfills it", async () => {
    await test("cross-fetch", alias, true)
  })
})
