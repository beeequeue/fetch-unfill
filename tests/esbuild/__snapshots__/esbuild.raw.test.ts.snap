// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`cross-fetch > bundles the original package > cjs 1`] = `"302.03KB"`;

exports[`cross-fetch > bundles the original package > mjs 1`] = `"302.72KB"`;

exports[`cross-fetch > unfills it > cjs 1`] = `"452B"`;

exports[`cross-fetch > unfills it > mjs 1`] = `"237B"`;

exports[`cross-fetch > unfills it > unfilled cjs code 1`] = `
""use strict";var r=(a,e)=>()=>(e||a((e={exports:{}}).exports,e),e.exports);var o=r((p,s)=>{"use strict";var l=globalThis.fetch;globalThis.fetch.polyfill=!1;l.polyfill=!1;s.exports=l;Object.assign(s.exports,{fetch:l,Request:globalThis.Request,Response:globalThis.Response,Headers:globalThis.Headers,AbortController:globalThis.AbortController})});var t=o();module.exports=()=>t("https://example.com");module.exports.example=()=>t("https://example.com");
"
`;

exports[`cross-fetch > unfills it > unfilled mjs code 1`] = `
"var t=globalThis.fetch;globalThis.fetch.polyfill=!1;t.polyfill=!1;var e=t,{Request:l,Response:o,Headers:s,AbortController:p}=globalThis;var c=()=>e("https://example.com"),h=()=>e("https://example.com");export{c as default,h as example};
"
`;

exports[`node-fetch > bundles the original package > cjs 1`] = `"92.2KB"`;

exports[`node-fetch > bundles the original package > mjs 1`] = `"90.41KB"`;

exports[`node-fetch > unfills it > cjs 1`] = `"452B"`;

exports[`node-fetch > unfills it > mjs 1`] = `"237B"`;

exports[`node-fetch > unfills it > unfilled cjs code 1`] = `
""use strict";var r=(a,e)=>()=>(e||a((e={exports:{}}).exports,e),e.exports);var o=r((p,s)=>{"use strict";var l=globalThis.fetch;globalThis.fetch.polyfill=!1;l.polyfill=!1;s.exports=l;Object.assign(s.exports,{fetch:l,Request:globalThis.Request,Response:globalThis.Response,Headers:globalThis.Headers,AbortController:globalThis.AbortController})});var t=o();module.exports=()=>t("https://example.com");module.exports.example=()=>t("https://example.com");
"
`;

exports[`node-fetch > unfills it > unfilled mjs code 1`] = `
"var t=globalThis.fetch;globalThis.fetch.polyfill=!1;t.polyfill=!1;var e=t,{Request:l,Response:o,Headers:s,AbortController:p}=globalThis;var c=()=>e("https://example.com"),h=()=>e("https://example.com");export{c as default,h as example};
"
`;
